playwright/tests/auth.setup.ts

```diff
@@ -1,22 +1,197 @@
-import { HomePage } from "@pages/home-page";
-import { LoginPage } from "@pages/login-page";
-import {
-  expect,
-  test as setup,
-} from "@playwright/test";
+import { USER_PERMISSION } from "@data/userPermissions";
+import { HomePage } from "@pages/homePage";
+import { LoginPage } from "@pages/loginPage";
+import { expect, test as setup } from "@playwright/test";
 
 const adminFile = "playwright/.auth/admin.json";
+const translationPermissionsFile = "playwright/.auth/translations.json";
+const productPermissionsFile = "playwright/.auth/product.json";
+const discountPermissionsFile = "playwright/.auth/discount.json";
+const settingsPermissionsFile = "playwright/.auth/settings.json";
+const staffMemberPermissionsFile = "playwright/.auth/staff-member.json";
+const ordersPermissionsFile = "playwright/.auth/orders.json";
+const shippingPermissionsFile = "playwright/.auth/shipping.json";
+const appsPermissionsFile = "playwright/.auth/apps.json";
+const pluginPermissionsFile = "playwright/.auth/plugins.json";
+const productTypePermissionsFile = "playwright/.auth/product-type.json";
+const giftCardsPermissionsFile = "playwright/.auth/gift-cards.json";
+const contentPermissionsFile = "playwright/.auth/content.json";
+const channelsWebhooksPermissionsFile =
+  "playwright/.auth/channels-webhooks.json";
+const customerWebhooksPermissionsFile = "playwright/.auth/customer.json";
 
 setup("authenticate as admin", async ({ page }) => {
   const loginPage = new LoginPage(page);
-  await loginPage.goto();
-  await loginPage.typeEmail(process.env.CYPRESS_USER_NAME!);
-  await loginPage.typePassword(process.env.CYPRESS_USER_PASSWORD!);
-  await loginPage.clickSignInButton();
-
+  await loginPage.setupLogin(
+    process.env.CYPRESS_USER_NAME!,
+    process.env.CYPRESS_USER_PASSWORD!,
+  );
   const homePage = new HomePage(page);
   await expect(homePage.welcomeMessage).toContainText("Hello there,");
-
   // End of authentication steps.
   await page.context().storageState({ path: adminFile });
 });
+setup("authenticate as user with discount permissions", async ({ page }) => {
+  const loginPage = new LoginPage(page);
+  await loginPage.setupLogin(
+    USER_PERMISSION.discount,
+    process.env.CYPRESS_PERMISSIONS_USERS_PASSWORD!,
+  );
+  const homePage = new HomePage(page);
+  await expect(homePage.welcomeMessage).toContainText("Hello there,");
+  // End of authentication steps.
+  await page.context().storageState({ path: discountPermissionsFile });
+});
+
+setup("authenticate as user with orders permissions", async ({ page }) => {
+  const loginPage = new LoginPage(page);
+  await loginPage.setupLogin(
+    USER_PERMISSION.order,
+    process.env.CYPRESS_PERMISSIONS_USERS_PASSWORD!,
+  );
+  const homePage = new HomePage(page);
+  await expect(homePage.welcomeMessage).toContainText("Hello there,");
+  // End of authentication steps.
+  await page.context().storageState({ path: ordersPermissionsFile });
+});
```

Comment:
This code is almost same for all setups, cant we create the function loginAndSetStorageState(esername, pastowrd, pathtostorageStateFile)